test/model/ReflectionTestHelper.java

```diff
@@ -0,0 +1,40 @@
+package model;
+
+import java.lang.reflect.Field;
+
+public class ReflectionTestHelper {
+    public static void changePrivateFieldOnObject(Object object, String fieldName,
+            Object newValue) {
+        try {
+            final Field field = object.getClass().getDeclaredField(fieldName);
+            field.setAccessible(true);
+            field.set(object, newValue);
+        } catch (final NoSuchFieldException e) {
+            e.printStackTrace();
+        } catch (final SecurityException e) {
+            e.printStackTrace();
+        } catch (final IllegalArgumentException e) {
+            e.printStackTrace();
+        } catch (final IllegalAccessException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    public static Object getPrivateFieldValue(Object object, String fieldName) {
+        try {
+            final Field field = object.getClass().getDeclaredField(fieldName);
+            field.setAccessible(true);
+            return field.get(object);
+        } catch (final NoSuchFieldException e) {
+            e.printStackTrace();
+        } catch (final SecurityException e) {
+            e.printStackTrace();
+        } catch (final IllegalArgumentException e) {
+            e.printStackTrace();
+        } catch (final IllegalAccessException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+}
```

Comment:
I don't believe so. It's not actually changing the field to public, rather suppressing the Java language access checking for the field. More info [here](http://docs.oracle.com/javase/7/docs/api/java/lang/reflect/AccessibleObject.html#setAccessible%28boolean%29).